extensions/itunes/init.lua

```diff
@@ -281,6 +281,20 @@ function itunes.setPosition(p)
   return tell('set player position to '..p)
 end
 
+--- hs.itunes.getDuration()
+--- Function
+--- Gets the duration (in seconds) of the current song
+---
+--- Parameters:
+---  * None
+---
+--- Returns:
+---  * The number of seconds long the current song is, 0 if no song is playing
+function itunes.getDuration()
+  local duration = tonumber(tell('duration of current track'))
+  return duration ~= nil and duration or 0
```

Comment:
No it's fine your explanation makes perfect sense... not sure what I was thinking. It's pretty late here so I misread the function as returning the current time of the song